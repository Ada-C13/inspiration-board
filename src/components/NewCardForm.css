/* The Modal (background) */

.new-card-form {
  width: 100%;
  padding-bottom: 4rem;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
}

.new-card-form__form {
  width: 50%;
  margin: auto;
  font-size: 1.5em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 1fr;
  grid-auto-flow: row;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-column: controls;
  grid-row: auto;
  margin-bottom: .8em;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid black;
  font-size: 1em;
}
