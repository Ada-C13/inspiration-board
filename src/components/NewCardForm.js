import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

const NewCardForm =(props) => {

  const emojiDropdown = () => {
    const emojiList = EMOJI_LIST.map(emoji=>{
      return  (<option value={emoji}>{emoji}</option>);
    });
    return emojiList;
  };



  return(

    <form className="NewCardForm" onSubmit={props.onSubmitCard}>
    <label>Card Text:</label>
     <input name = "cardtext" type="text" onChange={props.onChangeCard}/>
    <label>Emoji Input:</label>
     <select  name="cardemoji" onChange={props.onChangeCard}>
      {emojiDropdown()}
    </select>
    <input
        type="submit"
        value="Submit Card"
      />
    </form>
  );
};

export default NewCardForm;