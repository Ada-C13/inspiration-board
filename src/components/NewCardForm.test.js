import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'
import NewCardForm from './NewCardForm';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('NewCardForm', () => {
    test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <NewCardForm />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});

// describe('NewStudentForm', () => {
//   afterEach(cleanup); // Clear the DOM, to start with fresh empty HTML source code

//   const setup = () => {
//     const addStudentCallbackMock = jest.fn();
//     const renderResult = render(<NewStudentForm
//     addStudentCallback={addStudentCallbackMock}
//     />);

//     const fullNameInput = renderResult.getByLabelText(/Name:/i);
//     const emailInput = renderResult.getByLabelText(/Email:/i);
//     const submitButton = renderResult.getByText('Add Student');

//     return {
//       ...renderResult,
//       addStudentCallbackMock,
//       fullNameInput,
//       emailInput,
//       submitButton
//     };
//   }
//   test('that it matches the existing snapshot', () => {
//     // Arrange-Act
//     const { asFragment } = setup();

//     // Assert
//     expect(asFragment()).toMatchSnapshot();
//   });

//   test('Changing the fullName field changes the value', () => {
//     // Arrange
//     const { fullNameInput } = setup();

//     const simulatedEvent = {
//       target: {
//         value: 'Grace Hopper',
//       }
//     }

//     // Act
//     fireEvent.change(fullNameInput, simulatedEvent);

//     // Assert
//     expect(fullNameInput.value).toBe('Grace Hopper');
//   });

//   test('Changing the email field changes the value', () => {
//     // Arrange
//     const { emailInput } = setup();

//     const simulatedEvent = {
//       target: {
//         value: 'grace@navy.mil',
//       }
//     }

//     // Act
//     fireEvent.change(emailInput, simulatedEvent);

//     // Assert
//     expect(emailInput.value).toBe('grace@navy.mil');
//   });

//   test('Submitting the form calls the callback', () => {
//     // Arrange
//     const { emailInput, fullNameInput, getByTestId, addStudentCallbackMock } = setup();
//     const emailEvent = {
//       target: {
//         value: 'grace@navy.mil',
//       }
//     }
//     fireEvent.change(emailInput, emailEvent);

//     const fullNameEvent = {
//       target: {
//         value: 'Grace Hopper',
//       }
//     };
//     fireEvent.change(fullNameInput, fullNameEvent);

//     const form = getByTestId('new-student-form');
//     fireEvent.submit(form);

//     expect(addStudentCallbackMock).toHaveBeenCalled();
//     expect(addStudentCallbackMock).toHaveBeenCalledWith({
//       fullName: 'Grace Hopper',
//       email: 'grace@navy.mil',
//     });

//   })
// });
