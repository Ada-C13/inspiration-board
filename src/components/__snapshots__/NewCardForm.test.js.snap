// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <form
    class="new-card-form"
  >
    <header
      class="new-card-form__header"
    >
      Post a Note
    </header>
    <div
      class="new-card-form__form"
    >
      <label
        class="new-card-form__form-label"
        for="text"
      >
        Text
      </label>
      <input
        class="new-card-form__form-textarea "
        name="text"
        placeholder="Your text here"
        type="text"
        value=""
      />
      <label
        class="new-card-form__form-label"
        for="emoji"
      >
        Emoji
      </label>
      <input
        class="new-card-form__form-textarea"
        name="emoji"
        placeholder="Emoji name"
        type="text"
        value=""
      />
      <input
        class="new-card-form__form-button"
        type="submit"
        value="submit a note"
      />
    </div>
  </form>
</DocumentFragment>
`;
